doc/examples/adaptive_distances.ipynb

```diff
@@ -253,43 +284,256 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In detail, the adaptive distance feature works as follows: In each iteration of the ABCSMC run, after having obtained the desired number of accepted particles (and once at the beginning using a sample from the prior), the method ``DistanceFunction.update()`` is called. It is given a set of summary statistics which can be used to e.g. compute weights for the distance measure in the next iteration. In order to avoid bias, via ``DistanceFunction.configure_sampler()``, the distance function can tell the sampler to not only record accepted particles, but all that were generated during the sampling process.\n",
+    "So, when you want to define your own adaptive distance function, you will typically only need to overwrite these two methods. For implementation details and an example of how this can look in practice, please inspect the code of ``AdaptivePNormDistance``."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Make it robust\n",
+    "\n",
+    "A problem with the previous adaptive distance occurs when the weights do not work properly. E.g., there could be large weights assigned to rather non-informative statistics, if these vary comparably little. Let us have a look at an example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import scipy\n",
+    "import tempfile\n",
+    "import os\n",
+    "import matplotlib.pyplot as pyplot\n",
+    "import pyabc.visualization\n",
+    "import logging\n",
+    "\n",
+    "\n",
+    "# for debugging\n",
+    "df_logger = logging.getLogger('Distance')\n",
+    "df_logger.setLevel(logging.DEBUG)\n",
+    "\n",
+    "# model definition\n",
+    "def model(p):\n",
+    "    return {'ss1': p['theta'] + 1 + 1 * scipy.randn(),\n",
+    "            'ss2': 2 + 0.01 * scipy.randn()}\n",
+    "\n",
+    "# true model parameter\n",
+    "theta_true = 3\n",
+    "\n",
+    "# observed summary statistics\n",
+    "observation = {'ss1': theta_true + 1, 'ss2': 5}\n",
+    "\n",
+    "# prior distribution\n",
+    "prior = pyabc.Distribution(theta=pyabc.RV('uniform', 0, 10))\n",
+    "\n",
+    "# database\n",
+    "db_path = \"sqlite:///\" + os.path.join(tempfile.gettempdir(), \"tmp.db\")\n",
+    "\n",
+    "\n",
     "# plotting\n",
     "\n",
-    "fig, ax = pyplot.subplots()\n",
-    "for t in range(history1.max_t + 1):\n",
-    "    df, w = history1.get_distribution(m=0, t=t)\n",
-    "    pyabc.visualization.plot_kde_1d(df, w, xmin=0, xmax=10,\n",
-    "                                   x='theta', ax=ax,\n",
-    "                                   label=\"PDF t={}\".format(t))\n",
-    "ax.axvline(theta_true, color='k', linestyle='dashed', label=\"True value\")\n",
-    "ax.legend()"
+    "def plot(history):\n",
```

Comment:
this was, because the three subchapters define kind of independent things (still under the adaptive distances context, thus in the same notebook). so I wanted to have everything the user needs to run a subpart in the chapter.